### PR TITLE
Fix unsuccessful twin transfer

### LINKED ISSUE #651: Seller may lose deposit if they update the operator
If the following steps happen:
- a seller creeates an offer with bundle
- the seller updates the operator address
- the seller does not give approve the protocol to transfer the twins on behalf of the new operator
- a buyer with EOA redeems the voucher for this offer
then exchange goes directly into revoked state and the seller loses seller's deposit.

## Recommendation
If twin transfer fail, always automatically raise a dispute, regardless of what type the buyer is (EOA or contract)

### PR DESCRIPTION
**Pull request by zajck**
Fix #651

### CODE REVIEW

File: contracts/protocol/facets/ExchangeHandlerFacet.sol [hunk 1]
```diff
@@ -691,23 +685,18 @@ contract ExchangeHandlerFacet is IBosonExchangeHandler, BuyerBase, DisputeBase {
      * @notice Transfers bundled twins associated with an exchange to the buyer.
      *
      * Emits ERC20 Transfer, ERC721 Transfer, or ERC1155 TransferSingle events in call stack if successful.
+     * Emits TwinTransferred if twin transfer was successfull
+     * Emits TwinTransferFailed if twin transfer failed
      *
-     * Reverts if
-     * - A twin transfer fails
+     * If one of the twin transfers fails, the function will continue to transfer the remaining twins and
+     * autmaticaly raises a dispute for the exchange.
```
Comment (anajuliabit) [inline contracts/protocol/facets/ExchangeHandlerFacet.sol:692]:
```suggestion
     * automatically raises a dispute for the exchange.
```

**Review by anajuliabit — DISMISSED** [review]
Looks good

File: contracts/protocol/facets/ExchangeHandlerFacet.sol (hunk 1 above)
Comment (zajck) [reply contracts/protocol/facets/ExchangeHandlerFacet.sol]: Done.

**Review by anajuliabit — DISMISSED** [review]

File: contracts/protocol/facets/ExchangeHandlerFacet.sol [hunk 2]
```diff
@@ -808,17 +782,7 @@ contract ExchangeHandlerFacet is IBosonExchangeHandler, BuyerBase, DisputeBase {
...
-                if (sender.isContract()) {
-                    raiseDisputeInternal(_exchange, _voucher, seller.id);
-                } else {
-                    // Revoke voucher if caller is an EOA
-                    revokeVoucherInternal(_exchange);
-                    // N.B.: If voucher was revoked because transfer twin failed, then voucher was already burned
-                    shouldBurnVoucher = false;
-                }
-            }
+            if (transferFailed) raiseDisputeInternal(_exchange, _voucher, seller.id);
```
Comment (anajuliabit) [inline contracts/protocol/facets/ExchangeHandlerFacet.sol:785]: can you put this call to raiseDisputeInternal on line 770 and remove transferFailed variable pls

**Review by anajuliabit — CHANGES_REQUESTED** [review]

File: contracts/protocol/facets/ExchangeHandlerFacet.sol (hunk 2 above)
Comment (zajck) [reply contracts/protocol/facets/ExchangeHandlerFacet.sol]: Good point, nice one.

Changed.

**Review by anajuliabit — APPROVED** [review]
Looks good 👍

**Review by mischat — APPROVED** [review]
This looks good, the change and simplified it considerably, looks nicer now.

**Pull request merged**